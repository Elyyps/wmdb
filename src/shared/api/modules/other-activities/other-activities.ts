import { ILink } from "@app/api/core/link";
import { IEventCard } from "@app/api/core/event-card";

export interface IOtherActivities {
  button: ILink;
  cards: IEventCard[];
  title: string;
}
