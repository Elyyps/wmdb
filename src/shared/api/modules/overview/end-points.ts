import { IOverviewFilterItem, IOutingCard } from "./overview";
import { getCardsAmount } from "./dummy-data";
/* tslint:disable */
interface ICardsPaginatedReturn {
  cards: IOutingCard[];
  total: number;
}

export const getCardsPaginated = (skip: number, take: number, filter: IOverviewFilterItem): ICardsPaginatedReturn => {
  const CARDS_AMOUNT = 200;
  let cards = getCardsAmount(CARDS_AMOUNT);
  if (filter.filterText) {
    const filterTextCapitalized = filter.filterText.toUpperCase();
    cards = cards.filter(
      item =>
        item.title &&
        item.title
          .toUpperCase()
          .trim()
          .includes(filterTextCapitalized)
    );
  }
  if (filter.checkedItems.length > 0) {
    const checkedCategories = filter.checkedItems.filter(item => item.isChecked).map(item => item.id);
    cards = cards.filter(
      card => card.categoriesId && card.categoriesId.some(item => checkedCategories.indexOf(item) >= 0)
    );
  }
  if (filter.keyword) {
    const filterCapitalized = filter.keyword.toUpperCase();
    cards = cards.filter(
      card =>
        card.title.toUpperCase().includes(filterCapitalized) ||
        card.subtitle.toUpperCase().includes(filterCapitalized) ||
        card.content.toUpperCase().includes(filterCapitalized)
    );
  }

  if (filter.range) {
    cards = cards.filter(card => card.minimumPersons <= filter.range && card.maximumPersons >= filter.range);
  }

  const total = cards.length;
  cards = cards.slice(skip, skip + take);

  const cardAd: IOutingCard = {
    variant: "Ad",
    minimumPersons: 0,
    maximumPersons: 0,
    content: "",
    title: "Herfstspecial - bekijk nu de nieuwsbrief!",
    subtitle: "Citygames in Amsterdam",
    image: ["https://picsum.photos/id/103/1200/800"],
    button: {
      title: "Lees meer",
      href: "#"
    }
  };
  if (cards.length === 8) {
    cards = [...cards.slice(0, 2), cardAd, ...cards.slice(2, cards.length)];
  }

  return {
    cards,
    total
  };
};
