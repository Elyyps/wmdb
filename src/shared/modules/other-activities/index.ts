export * from "./other-activities.component";
