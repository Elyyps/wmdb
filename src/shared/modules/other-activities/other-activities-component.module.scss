.other-activities {
  > h2 {
    margin-bottom: 32px;
  }

  &-action {
    a {
      width: 100%;
    }
  }
}
