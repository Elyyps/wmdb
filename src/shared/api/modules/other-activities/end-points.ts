import { generateOtherActivitiesModuleData } from "./generate-dummy-data";

export const generateOtherActivitiesData = generateOtherActivitiesModuleData;
