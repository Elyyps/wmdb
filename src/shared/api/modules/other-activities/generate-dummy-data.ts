import { IOtherActivitiesModule } from "./other-activities";
import Image1 from "@assets/img-1.jpg";
import Image2 from "@assets/header-1.png";
import Image3 from "@assets/oasis.jpg";
import Image4 from "@assets/img-blog.jpg";

export const generateOtherActivitiesModuleData = (): IOtherActivitiesModule => ({
  buttonTitleClosed: "Bekijk 6 meer activiteiten",
  buttonTitleOpened: "Bekijk 6 minder activiteiten",
  numberItemsShowing: 4,
  title: "Andere activiteiten van dit bedrijf",
  cards: [
    { title: "U Deugt Wij Deugen", buttonText: "", image: Image1 },
    { title: "The Ultimate Expedition", buttonText: "", image: Image2 },
    { title: "Het Mislukte Feest", buttonText: "", image: Image3 },
    { title: "Hou van Holland Stadsspel", buttonText: "", image: Image4 },
    { title: "Het Mislukte Feest", buttonText: "", image: Image1 },
    { title: "Hou van Holland Stadsspel", buttonText: "", image: Image2 },
    { title: "Het Mislukte Feest", buttonText: "", image: Image3 },
    { title: "Hou van Holland Stadsspel", buttonText: "", image: Image4 },
    { title: "Het Mislukte Feest", buttonText: "", image: Image1 },
    { title: "Hou van Holland Stadsspel", buttonText: "", image: Image2 }
  ]
});
