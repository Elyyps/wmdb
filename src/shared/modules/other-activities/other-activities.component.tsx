import * as React from "react";
import styles from "./other-activities-component.module.scss";
import { EventCardComponent } from "@app/core/event-card";
import { Button } from "@app/prep/modules-prep/core";
import { IOtherActivities } from "@app/api/modules/other-activities/other-activities";
import { IEventCard } from "@app/api/core/event-card";

export interface IOtherActivitiesComponentProps {
  otherActivities: IOtherActivities;
}

const OtherActivitiesComponent = (props: IOtherActivitiesComponentProps) => (
  <div className={styles["other-activities"]}>
    <div className={styles["other-activities-head"]}>
      <h2>{props.otherActivities.title}</h2>
    </div>
    <div className="uk-grid uk-child-width-1-2@s uk-grid-medium" data-uk-margin>
      {props.otherActivities.cards &&
        props.otherActivities.cards.map((item: IEventCard, key: number) => (
          <div key={key}>
            <EventCardComponent image={item.image} link={item.buttonText} title={item.title} />
          </div>
        ))}
    </div>
    <div className={styles["other-activities-action"]}>
      <Button title={props.otherActivities.button.title} href={props.otherActivities.button.url} />
    </div>
  </div>
);

export { OtherActivitiesComponent };
