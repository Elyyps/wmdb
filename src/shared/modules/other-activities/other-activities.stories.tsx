import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { OtherActivitiesComponent } from "./other-activities.component";
import { generateOtherActivitiesData } from "@app/api/modules/other-activities/end-points";

storiesOf("OtherActivities", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => (
    <OtherActivitiesComponent otherActivitiesModule={generateOtherActivitiesData()} />
  ));
