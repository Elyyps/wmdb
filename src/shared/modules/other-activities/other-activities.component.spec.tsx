import React from "react";
import { shallow } from "enzyme";
import { OtherActivitiesComponent } from "./other-activities.component";
import { generateOtherActivitiesData } from "@app/api/modules/other-activities/end-points";

describe("[OtherActivities]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<OtherActivitiesComponent otherActivities={generateOtherActivitiesData()} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
