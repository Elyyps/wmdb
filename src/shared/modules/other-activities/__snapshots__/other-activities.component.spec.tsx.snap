// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[OtherActivities] should render component without crashing 1`] = `
<div
  className="other-activities"
>
  <h2>
    Andere activiteiten van dit bedrijf
  </h2>
  <div
    className="uk-grid uk-child-width-1-2@s"
  />
  <div
    className="other-activities-action"
  >
    <Button
      fullWidth={true}
      onClick={[Function]}
      title="Bekijk 6 meer activiteiten"
    />
  </div>
</div>
`;
